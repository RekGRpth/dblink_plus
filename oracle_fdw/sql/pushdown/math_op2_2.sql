\c contrib_regression_utf8

SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';

-- '-' operator
-- 180 : int2mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_smallint_1 = 0;
-- 181 : int4mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_integer_1 = 0;
-- 182 : int24mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_integer_1 = 0;
-- 183 : int42mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_smallint_1 = 0;
-- 205 : float4mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 - binary_float_real_1 = 0;
-- 206 : float4um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - binary_float_real_1 = -1;
-- 212 : int4um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_integer_1 = -1;
-- 213 : int2um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_smallint_1 = -1;
-- 219 : float8mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 - binary_double_precision_1 = 0;
-- 220 : float8um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - binary_double_precision_1 = -1;
-- 282 : float48mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 - binary_double_precision_1 = 0;
-- 286 : float84m
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 - binary_float_real_1 = 0;
-- 462 : int8um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_bigint_1 = -1;
-- 464 : int8mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_bigint_1 = 0;
-- 838 : int82mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_smallint_1 = 0;
-- 942 : int28mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_bigint_1 = 0;
-- 1275 : int84mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_integer_1 = 0;
-- 1279 : int48mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_bigint_1 = 0;
-- 1725 : numeric_sub
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 - number_numeric3_1 = 0;
-- 1771 : numeric_uminus
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_numeric3_1 = -1;

-- 180 : int2mi
SELECT id FROM all_type WHERE number_smallint_1 - number_smallint_1 = 0;
-- 181 : int4mi
SELECT id FROM all_type WHERE number_integer_1 - number_integer_1 = 0;
-- 182 : int24mi
SELECT id FROM all_type WHERE number_smallint_1 - number_integer_1 = 0;
-- 183 : int42mi
SELECT id FROM all_type WHERE number_integer_1 - number_smallint_1 = 0;
-- 205 : float4mi
SELECT id FROM all_type WHERE binary_float_real_1 - binary_float_real_1 = 0;
-- 206 : float4um
SELECT id FROM all_type WHERE - binary_float_real_1 = -1;
-- 212 : int4um
SELECT id FROM all_type WHERE - number_integer_1 = -1;
-- 213 : int2um
SELECT id FROM all_type WHERE - number_smallint_1 = -1;
-- 219 : float8mi
SELECT id FROM all_type WHERE binary_double_precision_1 - binary_double_precision_1 = 0;
-- 220 : float8um
SELECT id FROM all_type WHERE - binary_double_precision_1 = -1;
-- 282 : float48mi
SELECT id FROM all_type WHERE binary_float_real_1 - binary_double_precision_1 = 0;
-- 286 : float84m
SELECT id FROM all_type WHERE binary_double_precision_1 - binary_float_real_1 = 0;
-- 462 : int8um
SELECT id FROM all_type WHERE - number_bigint_1 = -1;
-- 464 : int8mi
SELECT id FROM all_type WHERE number_bigint_1 - number_bigint_1 = 0;
-- 838 : int82mi
SELECT id FROM all_type WHERE number_bigint_1 - number_smallint_1 = 0;
-- 942 : int28mi
SELECT id FROM all_type WHERE number_smallint_1 - number_bigint_1 = 0;
-- 1275 : int84mi
SELECT id FROM all_type WHERE number_bigint_1 - number_integer_1 = 0;
-- 1279 : int48mi
SELECT id FROM all_type WHERE number_integer_1 - number_bigint_1 = 0;
-- 1725 : numeric_sub
SELECT id FROM all_type WHERE number_numeric3_1 - number_numeric3_1 = 0;
-- 1771 : numeric_uminus
SELECT id FROM all_type WHERE - number_numeric3_1 = -1;

