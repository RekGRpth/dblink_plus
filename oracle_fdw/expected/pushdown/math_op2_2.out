\c contrib_regression_utf8
SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';
-- '-' operator
-- 180 : int2mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_smallint_1 = 0;
                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                          
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" - "number_smallint_1") = 0)
(2 rows)

-- 181 : int4mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_integer_1 = 0;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" - "number_integer_1") = 0)
(2 rows)

-- 182 : int24mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_integer_1 = 0;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                          
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" - "number_integer_1") = 0)
(2 rows)

-- 183 : int42mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_smallint_1 = 0;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                          
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" - "number_smallint_1") = 0)
(2 rows)

-- 205 : float4mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 - binary_float_real_1 = 0;
                                                                                                                                                                                                                                            QUERY PLAN                                                                                                                                                                                                                                            
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" - "binary_float_real_1") = 0)
(2 rows)

-- 206 : float4um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - binary_float_real_1 = -1;
                                                                                                                                                                                                                                  QUERY PLAN                                                                                                                                                                                                                                   
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "binary_float_real_1") = (-1))
(2 rows)

-- 212 : int4um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_integer_1 = -1;
                                                                                                                                                                                                                                 QUERY PLAN                                                                                                                                                                                                                                 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "number_integer_1") = (-1))
(2 rows)

-- 213 : int2um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_smallint_1 = -1;
                                                                                                                                                                                                                                 QUERY PLAN                                                                                                                                                                                                                                  
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "number_smallint_1") = (-1))
(2 rows)

-- 219 : float8mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 - binary_double_precision_1 = 0;
                                                                                                                                                                                                                                                  QUERY PLAN                                                                                                                                                                                                                                                  
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" - "binary_double_precision_1") = 0)
(2 rows)

-- 220 : float8um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - binary_double_precision_1 = -1;
                                                                                                                                                                                                                                     QUERY PLAN                                                                                                                                                                                                                                      
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "binary_double_precision_1") = (-1))
(2 rows)

-- 282 : float48mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 - binary_double_precision_1 = 0;
                                                                                                                                                                                                                                               QUERY PLAN                                                                                                                                                                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" - "binary_double_precision_1") = 0)
(2 rows)

-- 286 : float84m
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 - binary_float_real_1 = 0;
                                                                                                                                                                                                                                               QUERY PLAN                                                                                                                                                                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" - "binary_float_real_1") = 0)
(2 rows)

-- 462 : int8um
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_bigint_1 = -1;
                                                                                                                                                                                                                                QUERY PLAN                                                                                                                                                                                                                                 
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "number_bigint_1") = (-1))
(2 rows)

-- 464 : int8mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_bigint_1 = 0;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" - "number_bigint_1") = 0)
(2 rows)

-- 838 : int82mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_smallint_1 = 0;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" - "number_smallint_1") = 0)
(2 rows)

-- 942 : int28mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 - number_bigint_1 = 0;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" - "number_bigint_1") = 0)
(2 rows)

-- 1275 : int84mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 - number_integer_1 = 0;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                         
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" - "number_integer_1") = 0)
(2 rows)

-- 1279 : int48mi
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 - number_bigint_1 = 0;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                         
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" - "number_bigint_1") = 0)
(2 rows)

-- 1725 : numeric_sub
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 - number_numeric3_1 = 0;
                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                          
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_numeric3_1" - "number_numeric3_1") = 0)
(2 rows)

-- 1771 : numeric_uminus
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE - number_numeric3_1 = -1;
                                                                                                                                                                                                                                 QUERY PLAN                                                                                                                                                                                                                                  
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE ((- "number_numeric3_1") = (-1))
(2 rows)

-- 180 : int2mi
SELECT id FROM all_type WHERE number_smallint_1 - number_smallint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 181 : int4mi
SELECT id FROM all_type WHERE number_integer_1 - number_integer_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 182 : int24mi
SELECT id FROM all_type WHERE number_smallint_1 - number_integer_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 183 : int42mi
SELECT id FROM all_type WHERE number_integer_1 - number_smallint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 205 : float4mi
SELECT id FROM all_type WHERE binary_float_real_1 - binary_float_real_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 206 : float4um
SELECT id FROM all_type WHERE - binary_float_real_1 = -1;
 id 
----
  7
(1 row)

-- 212 : int4um
SELECT id FROM all_type WHERE - number_integer_1 = -1;
 id 
----
  7
(1 row)

-- 213 : int2um
SELECT id FROM all_type WHERE - number_smallint_1 = -1;
 id 
----
  7
(1 row)

-- 219 : float8mi
SELECT id FROM all_type WHERE binary_double_precision_1 - binary_double_precision_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 220 : float8um
SELECT id FROM all_type WHERE - binary_double_precision_1 = -1;
 id 
----
  7
(1 row)

-- 282 : float48mi
SELECT id FROM all_type WHERE binary_float_real_1 - binary_double_precision_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 286 : float84m
SELECT id FROM all_type WHERE binary_double_precision_1 - binary_float_real_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 462 : int8um
SELECT id FROM all_type WHERE - number_bigint_1 = -1;
 id 
----
  7
(1 row)

-- 464 : int8mi
SELECT id FROM all_type WHERE number_bigint_1 - number_bigint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 838 : int82mi
SELECT id FROM all_type WHERE number_bigint_1 - number_smallint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 942 : int28mi
SELECT id FROM all_type WHERE number_smallint_1 - number_bigint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 1275 : int84mi
SELECT id FROM all_type WHERE number_bigint_1 - number_integer_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 1279 : int48mi
SELECT id FROM all_type WHERE number_integer_1 - number_bigint_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 1725 : numeric_sub
SELECT id FROM all_type WHERE number_numeric3_1 - number_numeric3_1 = 0;
 id 
----
  5
  6
  7
(3 rows)

-- 1771 : numeric_uminus
SELECT id FROM all_type WHERE - number_numeric3_1 = -1;
 id 
----
  7
(1 row)

